restorers/dataloader/base/base_dataloader.py

```diff
@@ -28,46 +39,99 @@ def fetch_dataset(self, val_split: float, visualize_on_wandb: bool):
     def sanity_tests(self):
         raise NotImplementedError(f"{self.__class__.__name__ }.sanity_tests")
 
-    def read_image(self, image_path: str):
+    def read_image(self, image_path: str) -> tf.Tensor:
+        """
+        Function to read an image from the file path.
+
+        Args:
+            image_path (`str`): The image file path.
+
+        Returns:
+            tf.Tensor image in channel-last format.
+        """
         image = tf.io.read_file(image_path)
-        image = tf.image.decode_png(image, channels=3)
+        image = tf.io.decode_image(image, channels=3)
         image = tf.cast(image, dtype=tf.float32) / self.normalization_factor
         return image
 
-    def random_crop(self, input_image, gt_image):
-        input_image_shape = tf.shape(input_image)[:2]
-        crop_width = tf.random.uniform(
-            shape=(), maxval=input_image_shape[1] - self.image_size + 1, dtype=tf.int32
+    def random_crop(
+        self, low_light_image: tf.Tensor, enhanced_image: tf.Tensor
+    ) -> Tuple[tf.Tensor]:
+        """
+        Function to apply random cropping.
+
+        Args:
+            low_light_image (`tf.Tensor`): Low light image.
+            enhanced_image (`tf.Tensor`): Enhanced image.
+
+        Returns:
+            A tuple of random cropped image.
+        """
+        # Check whether the image size is smaller than the original image
+        image_size = tf.minimum(self.image_size, tf.shape(low_light_image)[0])
+
+        # Concatenate the low light and enhanced image
+        concatenated_image = tf.concat(
+            [low_light_image[None, ...], enhanced_image[None, ...]],
+            axis=0,  # Note the batch dims
```

Comment:
@ariG23498 Shouldn't the concatenation be on `axis=-1`?

Reply:
My rationale behind this code is:
- Including a batch dimension with `[None, ...]`
- And concatenating the images from the batch.

This makes more intuitive sense of how we would do a random crop.